homeassistant/components/nightscout/sensor.py

```diff
@@ -120,3 +162,61 @@ def _parse_icon(self) -> str:
     def extra_state_attributes(self):
         """Return the state attributes."""
         return self._attributes
+
+
+class Battery(CoordinatorEntity, SensorEntity):
+    """Battery sensor of Nightscout device."""
+
+    # SENSOR_ATTR = "battery_percentage_remaining"
+    _attr_device_class = DEVICE_CLASS_BATTERY
+    _attr_state_class = STATE_CLASS_MEASUREMENT
+    _attr_native_unit_of_measurement = PERCENTAGE
+    _attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
+
+    def __init__(self, coordinator, api: NightscoutAPI, name, unique_id):
+        """Initialize the Nightscout sensor."""
+        super().__init__(coordinator)
+        self.api = api
+        self._unique_id = unique_id
+        self._name = name
+        self._device_name = name
+        self._icon = "mdi:battery-unknown"
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the sensor."""
+        return self._unique_id
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def available(self):
+        """Return if the sensor data are available."""
+        return hasattr(self.coordinator.data[self._name], "uploader")
+
+    @property
+    def native_value(self):
+        """Return the state of the device."""
+        return self.coordinator.data[self._name].uploader.battery
+
+    @property
+    def icon(self) -> str:
+        """Battery state icon handling."""
+        return icon_for_battery_level(
+            battery_level=self.coordinator.data[self._name].uploader.battery,
+            charging=False,
+        )
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        uploader = self.coordinator.data[self._name].uploader
+        attr = {}
+        if hasattr(uploader, "type"):
+            attr[ATTR_TYPE] = uploader.type
+        if hasattr(uploader, "batteryVoltage"):
+            attr[ATTR_BATTERY_VOLTAGE] = uploader.batteryVoltage
```

Comment:
Please make a separate sensor entity for the voltage instead.

Reply:
I think I'll just remove it then. I see that this info is present on Nightscout's API docs, but none of my devices report this value. I added it as an attribute just in case.
I'll remove it and if someone later needs it, I can make another PR.